backend/worker/kafka_worker.go

```diff
@@ -228,6 +228,17 @@ func (k *KafkaBatchWorker) flushLogs(ctx context.Context) error {
 			if !strings.HasPrefix(logRow.Body, "ENOENT: no such file or directory") && !strings.HasPrefix(logRow.Body, "connect ECONNREFUSED") {
 				filteredRows = append(filteredRows, logRow)
 			}
+
+			if logRow.ServiceName != "" {
+				project, err := k.Worker.Resolver.Store.GetProject(ctx, int(logRow.ProjectId))
+				if err == nil && project != nil {
+					_, err := k.Worker.Resolver.Store.FindOrCreateService(ctx, *project, logRow.ServiceName, logRow.LogAttributes)
+
+					if err != nil {
+						log.WithContext(ctx).Error(e.Wrap(err, "failed to create service"))
+					}
+				}
+			}
```

Comment:
we could we wrap this section in a trace to make sure we can isolate the performance of the `GetProject` + `FindOrCreateService`